Fix Between validator behavior

codisart opened the pull request:
fixes #167 

return false when validating a string with a numeric range
return false when validating a numeric value with a string range

Commit:
Corrected Date validator format docs

codisart committed:
Fix Between validator behavior

return false when validating a string with a numeric range
return false when validating a numeric value with a string range

samsonasik commented on src/Between.php:
```diff
@@ -17,6 +17,8 @@ class Between extends AbstractValidator
     const NOT_BETWEEN        = 'notBetween';
     const NOT_BETWEEN_STRICT = 'notBetweenStrict';
 
+    private $nuumeric;
```
typo, `nuumeric` should be `numeric`, surplus `u` char

codisart replied:
Thank you for you review @samsonasik !

froschdesign commented on src/Between.php:
```diff
@@ -17,6 +17,8 @@ class Between extends AbstractValidator
     const NOT_BETWEEN        = 'notBetween';
     const NOT_BETWEEN_STRICT = 'notBetweenStrict';
 
+    private $numeric;
```
Please add always a DocBlock with a short description and data type.

codisart replied:
Thank you for your review @froschdesign

Commit:
Fix BC break introduced by 3fc1e66

weierophinney committed:
Merge pull request #173 from nathanjosiah/fix-158

Corrected Date validator format docs

weierophinney committed:
Merge branch 'hotfix/173'

Close #173

weierophinney commented on src/Between.php:
```diff
@@ -162,6 +179,13 @@ public function setInclusive($inclusive)
+        }
+        if (! $this->numeric && ! is_string($value)) {
+            return false;
+        }
```
Each of these should be calling `$this->error()` in order to set a validation failure message; otherwise, there's no way to understand _why_ validation failed. Take a look at how the other cases that result in failure work, if you are unsure how to proceed.

weierophinney replied:
Also, once you do make this change, add tests for each condition to validate that the expected failure messages are present.

weierophinney commented on test/BetweenTest.php:
```diff
@@ -18,37 +18,76 @@
+                'max' => 100,
+                'inclusive' => true,
+                'expected' => true,
+                'values' => [1, 10, 100],
```
If you're going to break this into a data provider, do a separate set per value as well.

Additionally, give each set a key with a descriptive name:

```php
'inclusive-valid-floor' => [
    'min' => 1,
    'max' => 100,
    'inclusive' => true,
    'expected' => true,
    'value' => 1,
],
'inclusive-valid-in-range' => [
    'min' => 1,
    'max' => 100,
    'inclusive' => true,
    'expected' => true,
    'value' => 1,
],
'inclusive-valid-ceiling' => [
    'min' => 1,
    'max' => 100,
    'inclusive' => true,
    'expected' => true,
    'value' => 100,
],
```

This makes it easy to identify which case to examine when a failure occurs, as PHPUnit will report the key; you can then search for that key to identify each set of data provided.

weierophinney reviewed (changes requested):
This looks good, but needs a few changes before I can merge it.

weierophinney committed:
Updates Hostname IDN list

weierophinney committed:
Merge branch 'hotfix/hostname-idn-update'

Updates Hostname validator IDN list

weierophinney committed:
Merge pull request #180 from ThomasCantonnet/finfo-bc-break

Fix BC break introduced by 3fc1e66

weierophinney committed:
Adds CHANGELOG for #180

weierophinney committed:
Merge branch 'hotfix/180'

Close #180
Fixes #170
Fixes #176

weierophinney committed:
Updates Travis configuration to remove unneeded functionality

- Removes targets with SERVICEMANAGER; these are covered in the
  lowest/locked/latest strategy.
- Removes documentation deployment functionality; happens via github
  hooks now.
- Removes slack notification; happens via github hooks now.

weierophinney committed:
Merge branch 'hotfix/travis'

Updates travis to remove unneeded functionality.

weierophinney committed:
Merge pull request #174 from codisart/issue-167

Fix Between validator behavior

weierophinney committed:
Updates `providerBasic` to do one value per set

Splits each "values" argument into discrete sets. Names each set based
on behavior being tested.

weierophinney committed:
Provides validation error messages for Between validator type mismatch

Adds two new validation error messages, one for when a numeric value is
validated against non-numeric min/max values, and one for when a string
value is validated against numeric min/max values.

weierophinney committed:
Added CHANGELOG for #174

weierophinney commented:
@codisart I've made the changes I requested, and pushed them to your branch so as to verify the pull request against Travis. Merging now so we can do a new 2.9 series release.

Thanks!
